Kernel/FileSystem/ProcFS.cpp

```diff
@@ -593,18 +593,18 @@ Optional<KBuffer> procfs$pid_regs(InodeIdentifier identifier)
     KBufferBuilder builder;
     process.for_each_thread([&](Thread& thread) {
         builder.appendf("Thread %d:\n", thread.tid());
-        auto& tss = thread.tss();
-        builder.appendf("eax: %x\n", tss.eax);
-        builder.appendf("ebx: %x\n", tss.ebx);
-        builder.appendf("ecx: %x\n", tss.ecx);
-        builder.appendf("edx: %x\n", tss.edx);
-        builder.appendf("esi: %x\n", tss.esi);
-        builder.appendf("edi: %x\n", tss.edi);
-        builder.appendf("ebp: %x\n", tss.ebp);
-        builder.appendf("cr3: %x\n", tss.cr3);
-        builder.appendf("flg: %x\n", tss.eflags);
-        builder.appendf("sp:  %w:%x\n", tss.ss, tss.esp);
-        builder.appendf("pc:  %w:%x\n", tss.cs, tss.eip);
+        auto& regs = thread.get_register_dump_from_stack();
+        builder.appendf("eax: %x\n", regs.eax);
+        builder.appendf("ebx: %x\n", regs.ebx);
+        builder.appendf("ecx: %x\n", regs.ecx);
+        builder.appendf("edx: %x\n", regs.edx);
+        builder.appendf("esi: %x\n", regs.esi);
+        builder.appendf("edi: %x\n", regs.edi);
+        builder.appendf("ebp: %x\n", regs.ebp);
+        builder.appendf("cr3: %x\n", thread.cr3());
+        builder.appendf("flg: %x\n", regs.eflags);
+        builder.appendf("sp:  %w:%x\n", regs.ss, regs.esp);
+        builder.appendf("pc:  %w:%x\n", regs.cs, regs.eip);
```

Comment:
Can we please have this as JSON?

I understand you're just porting it to register dumps, but uhhh